src/main.ts

```diff
@@ -194,9 +215,13 @@ async function start(): Promise<void> {
   datasetSelectEl.addEventListener("change", handleDatasetChange);
   featureSelectEl.addEventListener("change", handleFeatureChange);
   colorRampSelectEl.addEventListener("click", handleColorRampClick);
+  canv.domElement.addEventListener("click", handleCanvasClick);
 }
 
-window.addEventListener("beforeunload", () => canv.dispose());
+window.addEventListener("beforeunload", () => {
+  canv.domElement.removeEventListener("click", handleCanvasClick);
+  canv.dispose();
+});
```

Comment:
I'm honestly not certain that I needed to add this event handler in the first place